Dockerfile

```diff
@@ -35,33 +33,27 @@ jdk.management.agent,\
 java.naming,jdk.naming.dns,\
 # TLS handehake with servers that use elliptic curve certificates
 jdk.crypto.ec,\
-# sun.misc.Unsafe and friends
+# jdk.unsupported is undocumented but contains Unsafe, which is used by several dependencies to
+# improve performance. Ex. sun.misc.Unsafe and friends
 jdk.unsupported,\
 # Elasticsearch 7+ crashes without Thai Segmentation support
 #  Add <900K instead of a different base layer
 jdk.localedata --include-locales en,th\
  --output jre
 
-# We extract JRE's hard dependencies, libz and SSL certs, from the fat JRE image.
-#   See https://console.cloud.google.com/gcr/images/distroless/GLOBAL/java
-FROM gcr.io/distroless/java:11-debug AS deps
-
-# Mainly, this gets BusyBox
-#   See https://console.cloud.google.com/gcr/images/distroless/GLOBAL/cc
-FROM gcr.io/distroless/cc:debug
-
-LABEL MAINTAINER Zipkin "https://zipkin.io/"
+FROM alpine:3.12
 
-# Similar to Alpine Linux, we ensure /bin/sh works (via BusyBox)
-RUN ["/busybox/sh", "-c", "ln -s /busybox/sh /bin/sh"]
+MAINTAINER OpenZipkin "http://zipkin.io/"
 
-COPY --from=deps /etc/ssl/certs/java /etc/ssl/certs/java
-COPY --from=deps /lib/x86_64-linux-gnu/libz.so.1.2.8 /lib/x86_64-linux-gnu/libz.so.1.2.8
-RUN ln -s /lib/x86_64-linux-gnu/libz.so.1.2.8 /lib/x86_64-linux-gnu/libz.so.1
+# Default to UTF-8 file.encoding
+ENV LANG C.UTF-8
 
-COPY --from=jre /jre /jre
+# Java relies on /etc/nsswitch.conf. Put host files first or InetAddress.getLocalHost
```

Comment:
Is this something special because of alpine? Wasn't expecting anything new :) Also, if zulu-alpine already does it, can we just copy out of it?

Reply:
This is from our JDK 1.8 image which is alpine based... so I didn't want to introduce a regression.

Reply:
here's our commit which was against an alpine docker image at that point, and carried forward on all alpine since.. https://github.com/openzipkin/docker-jre-full/commit/18049c9e607cf0e23d20e7c4552fd2a9b58b3b18#diff-bffaae65aa3c12762f8a3bca694d875b

I'll see if zulu alpine also does this

Reply:
there is no /etc/nsswitch.conf in azul/zulu-openjdk-alpine:15

Reply:
seems even DDG can find numerous new images adding nsswitch.conf to alpine dists, so I'll leave it if ok with you given the background